### PR TITLE
Dynamically determine external urls from sling maps

### PR DESCRIPTION
**Pull request by wrandall22**
This PR seeks to make the content scoring code a bit more maintainable by not having to hard-code externalizer configurations in the xml. Instead, this will loop through the existing sling maps (under `/etc/map.publish.environment`) until it finds a path that matches the one coming in. When it finds one, it will use that host/domain to build the external URI.

I feel like this is probably in existence somewhere inside of AEM's internal code, but, as far as I can tell, it's not exposed in OSGi and therefore inaccessible.

This work originally came about from [this Rollbar issue](https://rollbar.com/Cru/cru-aem-content-scoring/items/10/). Though, for now on the server I have added the externalizers manually.

### CODE REVIEW

File: core/src/main/java/org/cru/contentscoring/core/servlets/ResourceUrlMapperServlet.java [hunk 1]
```diff
@@ -83,4 +85,67 @@ protected void doGet(final SlingHttpServletRequest request, final SlingHttpServl
...
+
+                ValueMap properties = child.getValueMap();
+                String internalRedirect = properties.get("sling:internalRedirect", String.class);
+                if (internalRedirect != null && path.startsWith(internalRedirect)) {
+                    return child;
+                }
+            }
+        }
+        return null;
+    }
```
Comment (mattdrees) [inline core/src/main/java/org/cru/contentscoring/core/servlets/ResourceUrlMapperServlet.java]: > I feel like I am recreating something AEM does out of the box, but I cannot find a way to plug into that functionality directly

Did you find out which class does this, at least?

A little googling led me to https://sling.apache.org/documentation/the-sling-engine/mappings-for-resource-resolution.html#mapping-entry-specification and then https://svn.apache.org/repos/asf/sling/tags/org.apache.sling.jcr.resource-2.0.8/src/main/java/org/apache/sling/jcr/resource/internal/JcrResourceResolverFactoryImpl.java, but that class didn't seem to be the one that used the mapRoot; something else must be calling getMapRoot() and doing the the kind of work you're doing.

Oh well. Probably not worth more digging.

**Review by mattdrees — APPROVED** [review]
definitely worth removing that redundant config.

File: core/src/main/java/org/cru/contentscoring/core/servlets/ResourceUrlMapperServlet.java (hunk 1 above)
Comment (wrandall22) [reply core/src/main/java/org/cru/contentscoring/core/servlets/ResourceUrlMapperServlet.java]: I think the key class might be [this one](https://github.com/apache/sling-org-apache-sling-resourceresolver/blob/master/src/main/java/org/apache/sling/resourceresolver/impl/mapping/MapEntries.java)

**Comment by wrandall22** [comment]
Appears to have caused [this Rollbar issue](https://rollbar.com/Cru/cru-aem-content-scoring/items/26/) so I'll need to do a little more work.

**Pull request merged**